cms/static/js/views/utils/create_course_utils.js

```diff
@@ -1,57 +1,18 @@
 /**
  * Provides utilities for validating courses during creation, for both new courses and reruns.
  */
-define(["jquery", "underscore", "gettext", "js/views/utils/view_utils"],
-    function ($, _, gettext, ViewUtils) {
+define(["jquery", "gettext", "js/views/utils/view_utils", "js/views/utils/create_utils_base"],
+    function ($, gettext, ViewUtils, CreateUtilsFactory) {
+        "use strict";
         return function (selectors, classes) {
-            var validateTotalCourseItemsLength, setNewCourseFieldInErr, hasInvalidRequiredFields,
-                createCourse, validateFilledFields, configureHandlers;
+            CreateUtilsFactory.call(this, selectors, classes);
```

Comment:
@cahrens I found that the build was failing and so came back to this. JavaScript seems to love surprising me.

Assigning the prototype did allow the functions to become available, but it did not properly rescope `this` during the construction, which meant that use of the `self` variable within the CreateUtilsFactory functions still understood `this` to be CreateUtilsFactory and not CreateCourseUtilsFactory.

I decided to use `call` instead of `apply` here. I looked it up and `apply` is just call but expects the second argument to be an array. This would explain why it never worked, since I was handing it an object, but expecting it to be handled as one unit. JavaScript threw up and made `selector` and `classes` undefined, because it's better to plow ahead even when uncertain, apparently. :P

Anyway, this solution does allow `this` to be scoped properly. The prototype doesn't need to be set here, but using call takes care of everything, things look right and the test passes. As I no longer need to refer to the function, I'm just returning it directly instead of naming it, as well.